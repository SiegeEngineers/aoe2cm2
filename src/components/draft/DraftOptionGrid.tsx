import * as React from 'react';
import DraftOptionPanel from "../../containers/DraftOptionPanel";
import DraftOptionPanelType from "../../constants/DraftOptionPanelType";
import DraftOption from "../../models/DraftOption";

interface IProps {
    draftOptions: DraftOption[]
}

class DraftOptionGrid extends React.Component<IProps, object> {
    public render() {

        const panels = this.props.draftOptions.map((draftOption, index) => {
            return React.createElement(DraftOptionPanel, {
                draftOptionPanelType: DraftOptionPanelType.CHOICE,
                active: false,
                draftOption: draftOption,
                key: index
            });
        });

        const randomOption = React.createElement(DraftOptionPanel, {
            draftOptionPanelType: DraftOptionPanelType.CHOICE,
            active: false,
            draftOption: DraftOption.RANDOM,
            key: -1
        });

        return (
            <div id="civgrid" className="chooser">
                <div className="chooser-grid">
                    {randomOption}
                    {panels}
                </div>
            </div>
        );
    }
}

export default DraftOptionGrid;
