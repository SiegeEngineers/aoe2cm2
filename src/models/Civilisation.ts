import GameVersion from "../constants/GameVersion";
import DraftOption from "./DraftOption";

enum Name {
    BRITONS = "Britons",
    BYZANTINES = "Byzantines",
    CELTS = "Celts",
    CHINESE = "Chinese",
    FRANKS = "Franks",
    GOTHS = "Goths",
    JAPANESE = "Japanese",
    MONGOLS = "Mongols",
    PERSIANS = "Persians",
    SARACENS = "Saracens",
    TEUTONS = "Teutons",
    TURKS = "Turks",
    VIKINGS = "Vikings",

    AZTECS = "Aztecs",
    HUNS = "Huns",
    KOREANS = "Koreans",
    MAYANS = "Mayans",
    SPANISH = "Spanish",

    ITALIANS = "Italians",
    INCAS = "Incas",
    INDIANS = "Indians",
    MAGYARS = "Magyars",
    SLAVS = "Slavs",

    BERBERS = "Berbers",
    ETHIOPIANS = "Ethiopians",
    MALIANS = "Malians",
    PORTUGUESE = "Portuguese",

    BURMESE = "Burmese",
    KHMER = "Khmer",
    MALAY = "Malay",
    VIETNAMESE = "Vietnamese",

    BULGARIANS = "Bulgarians",
    CUMANS = "Cumans",
    LITHUANIANS = "Lithuanians",
    TATARS = "Tatars",

    BURGUNDIANS = "Burgundians",
    SICILIANS = "Sicilians",

    BOHEMIANS = "Bohemians",
    POLES = "Poles",
}

class Civilisation extends DraftOption {

    public static readonly BRITONS: Civilisation = new Civilisation(Name.BRITONS, GameVersion.AOK);
    public static readonly BYZANTINES: Civilisation = new Civilisation(Name.BYZANTINES, GameVersion.AOK);
    public static readonly CELTS: Civilisation = new Civilisation(Name.CELTS, GameVersion.AOK);
    public static readonly CHINESE: Civilisation = new Civilisation(Name.CHINESE, GameVersion.AOK);
    public static readonly FRANKS: Civilisation = new Civilisation(Name.FRANKS, GameVersion.AOK);
    public static readonly GOTHS: Civilisation = new Civilisation(Name.GOTHS, GameVersion.AOK);
    public static readonly JAPANESE: Civilisation = new Civilisation(Name.JAPANESE, GameVersion.AOK);
    public static readonly MONGOLS: Civilisation = new Civilisation(Name.MONGOLS, GameVersion.AOK);
    public static readonly PERSIANS: Civilisation = new Civilisation(Name.PERSIANS, GameVersion.AOK);
    public static readonly SARACENS: Civilisation = new Civilisation(Name.SARACENS, GameVersion.AOK);
    public static readonly TEUTONS: Civilisation = new Civilisation(Name.TEUTONS, GameVersion.AOK);
    public static readonly TURKS: Civilisation = new Civilisation(Name.TURKS, GameVersion.AOK);
    public static readonly VIKINGS: Civilisation = new Civilisation(Name.VIKINGS, GameVersion.AOK);

    public static readonly AZTECS: Civilisation = new Civilisation(Name.AZTECS, GameVersion.CONQUERORS);
    public static readonly HUNS: Civilisation = new Civilisation(Name.HUNS, GameVersion.CONQUERORS);
    public static readonly KOREANS: Civilisation = new Civilisation(Name.KOREANS, GameVersion.CONQUERORS);
    public static readonly MAYANS: Civilisation = new Civilisation(Name.MAYANS, GameVersion.CONQUERORS);
    public static readonly SPANISH: Civilisation = new Civilisation(Name.SPANISH, GameVersion.CONQUERORS);

    public static readonly ITALIANS: Civilisation = new Civilisation(Name.ITALIANS, GameVersion.FORGOTTEN);
    public static readonly INCAS: Civilisation = new Civilisation(Name.INCAS, GameVersion.FORGOTTEN);
    public static readonly INDIANS: Civilisation = new Civilisation(Name.INDIANS, GameVersion.FORGOTTEN);
    public static readonly MAGYARS: Civilisation = new Civilisation(Name.MAGYARS, GameVersion.FORGOTTEN);
    public static readonly SLAVS: Civilisation = new Civilisation(Name.SLAVS, GameVersion.FORGOTTEN);

    public static readonly BERBERS: Civilisation = new Civilisation(Name.BERBERS, GameVersion.AFRICAN_KINGDOMS);
    public static readonly ETHIOPIANS: Civilisation = new Civilisation(Name.ETHIOPIANS, GameVersion.AFRICAN_KINGDOMS);
    public static readonly MALIANS: Civilisation = new Civilisation(Name.MALIANS, GameVersion.AFRICAN_KINGDOMS);
    public static readonly PORTUGUESE: Civilisation = new Civilisation(Name.PORTUGUESE, GameVersion.AFRICAN_KINGDOMS);

    public static readonly BURMESE: Civilisation = new Civilisation(Name.BURMESE, GameVersion.RISE_OF_RAJAS);
    public static readonly KHMER: Civilisation = new Civilisation(Name.KHMER, GameVersion.RISE_OF_RAJAS);
    public static readonly MALAY: Civilisation = new Civilisation(Name.MALAY, GameVersion.RISE_OF_RAJAS);
    public static readonly VIETNAMESE: Civilisation = new Civilisation(Name.VIETNAMESE, GameVersion.RISE_OF_RAJAS);

    public static readonly BULGARIANS: Civilisation = new Civilisation(Name.BULGARIANS, GameVersion.DEFINITIVE_EDITION);
    public static readonly CUMANS: Civilisation = new Civilisation(Name.CUMANS, GameVersion.DEFINITIVE_EDITION);
    public static readonly LITHUANIANS: Civilisation = new Civilisation(Name.LITHUANIANS, GameVersion.DEFINITIVE_EDITION);
    public static readonly TATARS: Civilisation = new Civilisation(Name.TATARS, GameVersion.DEFINITIVE_EDITION);

    public static readonly BURGUNDIANS: Civilisation = new Civilisation(Name.BURGUNDIANS, GameVersion.LORDS_OF_THE_WEST);
    public static readonly SICILIANS: Civilisation = new Civilisation(Name.SICILIANS, GameVersion.LORDS_OF_THE_WEST);

    public static readonly BOHEMIANS: Civilisation = new Civilisation(Name.BOHEMIANS, GameVersion.DAWN_OF_THE_DUKES);
    public static readonly POLES: Civilisation = new Civilisation(Name.POLES, GameVersion.DAWN_OF_THE_DUKES);


    // DO NOT CHANGE THE ORDER OF ELEMENTS IN THIS ARRAY!!!
    // ONLY APPEND NEW CIVILISATIONS AT THE END!!!
    public static readonly ALL = [
        Civilisation.AZTECS,
        Civilisation.BERBERS,
        Civilisation.BRITONS,
        Civilisation.BURMESE,
        Civilisation.BYZANTINES,
        Civilisation.CELTS,
        Civilisation.CHINESE,
        Civilisation.ETHIOPIANS,
        Civilisation.FRANKS,
        Civilisation.GOTHS,
        Civilisation.HUNS,
        Civilisation.INCAS,
        Civilisation.INDIANS,
        Civilisation.ITALIANS,
        Civilisation.JAPANESE,
        Civilisation.KHMER,
        Civilisation.KOREANS,
        Civilisation.MAGYARS,
        Civilisation.MALAY,
        Civilisation.MALIANS,
        Civilisation.MAYANS,
        Civilisation.MONGOLS,
        Civilisation.PERSIANS,
        Civilisation.PORTUGUESE,
        Civilisation.SARACENS,
        Civilisation.SLAVS,
        Civilisation.SPANISH,
        Civilisation.TEUTONS,
        Civilisation.TURKS,
        Civilisation.VIETNAMESE,
        Civilisation.VIKINGS,
        Civilisation.BULGARIANS,
        Civilisation.CUMANS,
        Civilisation.LITHUANIANS,
        Civilisation.TATARS,
        Civilisation.BURGUNDIANS,
        Civilisation.SICILIANS,
        Civilisation.BOHEMIANS,
        Civilisation.POLES,
    ];

    public readonly gameVersion: GameVersion;

    private constructor(name: Name, gameVersion: GameVersion) {
        super(name);
        this.gameVersion = gameVersion;
    }

}

export default Civilisation;
export {Name};
