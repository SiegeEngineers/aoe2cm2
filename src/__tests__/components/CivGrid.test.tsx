import {shallow} from "enzyme";
import * as React from "react";
import DraftOptionGrid from "../../components/draft/DraftOptionGrid";
import Civilisation from "../../models/Civilisation";

it('DraftOptionGrid renders correctly', () => {
    const component = shallow(<DraftOptionGrid draftOptions={Civilisation.ALL}/>);
    expect(component).toMatchSnapshot();
});
