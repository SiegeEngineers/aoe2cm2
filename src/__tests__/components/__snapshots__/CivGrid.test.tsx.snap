// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DraftOptionGrid renders correctly 1`] = `
<div
  className="chooser"
  id="civgrid"
>
  <div
    className="chooser-grid"
  >
    <Connect(Component)
      active={false}
      draftOption={
        DraftOption {
          "id": "RANDOM",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/random-left.apng",
            "animated_right": "/images/units-animated/random-right.apng",
            "emblem": "/images/civemblems/random.png",
            "unit": "/images/civs/random.png",
          },
          "name": "RANDOM",
        }
      }
      draftOptionPanelType="choice"
      key="-1"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 2,
          "id": "Aztecs",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/aztecs-left.apng",
            "animated_right": "/images/units-animated/aztecs-right.apng",
            "emblem": "/images/civemblems/aztecs.png",
            "unit": "/images/civs/aztecs.png",
          },
          "name": "Aztecs",
        }
      }
      draftOptionPanelType="choice"
      key="0"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 4,
          "id": "Berbers",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/berbers-left.apng",
            "animated_right": "/images/units-animated/berbers-right.apng",
            "emblem": "/images/civemblems/berbers.png",
            "unit": "/images/civs/berbers.png",
          },
          "name": "Berbers",
        }
      }
      draftOptionPanelType="choice"
      key="1"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 1,
          "id": "Britons",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/britons-left.apng",
            "animated_right": "/images/units-animated/britons-right.apng",
            "emblem": "/images/civemblems/britons.png",
            "unit": "/images/civs/britons.png",
          },
          "name": "Britons",
        }
      }
      draftOptionPanelType="choice"
      key="2"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 5,
          "id": "Burmese",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/burmese-left.apng",
            "animated_right": "/images/units-animated/burmese-right.apng",
            "emblem": "/images/civemblems/burmese.png",
            "unit": "/images/civs/burmese.png",
          },
          "name": "Burmese",
        }
      }
      draftOptionPanelType="choice"
      key="3"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 1,
          "id": "Byzantines",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/byzantines-left.apng",
            "animated_right": "/images/units-animated/byzantines-right.apng",
            "emblem": "/images/civemblems/byzantines.png",
            "unit": "/images/civs/byzantines.png",
          },
          "name": "Byzantines",
        }
      }
      draftOptionPanelType="choice"
      key="4"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 1,
          "id": "Celts",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/celts-left.apng",
            "animated_right": "/images/units-animated/celts-right.apng",
            "emblem": "/images/civemblems/celts.png",
            "unit": "/images/civs/celts.png",
          },
          "name": "Celts",
        }
      }
      draftOptionPanelType="choice"
      key="5"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 1,
          "id": "Chinese",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/chinese-left.apng",
            "animated_right": "/images/units-animated/chinese-right.apng",
            "emblem": "/images/civemblems/chinese.png",
            "unit": "/images/civs/chinese.png",
          },
          "name": "Chinese",
        }
      }
      draftOptionPanelType="choice"
      key="6"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 4,
          "id": "Ethiopians",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/ethiopians-left.apng",
            "animated_right": "/images/units-animated/ethiopians-right.apng",
            "emblem": "/images/civemblems/ethiopians.png",
            "unit": "/images/civs/ethiopians.png",
          },
          "name": "Ethiopians",
        }
      }
      draftOptionPanelType="choice"
      key="7"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 1,
          "id": "Franks",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/franks-left.apng",
            "animated_right": "/images/units-animated/franks-right.apng",
            "emblem": "/images/civemblems/franks.png",
            "unit": "/images/civs/franks.png",
          },
          "name": "Franks",
        }
      }
      draftOptionPanelType="choice"
      key="8"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 1,
          "id": "Goths",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/goths-left.apng",
            "animated_right": "/images/units-animated/goths-right.apng",
            "emblem": "/images/civemblems/goths.png",
            "unit": "/images/civs/goths.png",
          },
          "name": "Goths",
        }
      }
      draftOptionPanelType="choice"
      key="9"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 2,
          "id": "Huns",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/huns-left.apng",
            "animated_right": "/images/units-animated/huns-right.apng",
            "emblem": "/images/civemblems/huns.png",
            "unit": "/images/civs/huns.png",
          },
          "name": "Huns",
        }
      }
      draftOptionPanelType="choice"
      key="10"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 3,
          "id": "Incas",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/incas-left.apng",
            "animated_right": "/images/units-animated/incas-right.apng",
            "emblem": "/images/civemblems/incas.png",
            "unit": "/images/civs/incas.png",
          },
          "name": "Incas",
        }
      }
      draftOptionPanelType="choice"
      key="11"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 3,
          "id": "Indians",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/indians-left.apng",
            "animated_right": "/images/units-animated/indians-right.apng",
            "emblem": "/images/civemblems/indians.png",
            "unit": "/images/civs/indians.png",
          },
          "name": "Indians",
        }
      }
      draftOptionPanelType="choice"
      key="12"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 3,
          "id": "Italians",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/italians-left.apng",
            "animated_right": "/images/units-animated/italians-right.apng",
            "emblem": "/images/civemblems/italians.png",
            "unit": "/images/civs/italians.png",
          },
          "name": "Italians",
        }
      }
      draftOptionPanelType="choice"
      key="13"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 1,
          "id": "Japanese",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/japanese-left.apng",
            "animated_right": "/images/units-animated/japanese-right.apng",
            "emblem": "/images/civemblems/japanese.png",
            "unit": "/images/civs/japanese.png",
          },
          "name": "Japanese",
        }
      }
      draftOptionPanelType="choice"
      key="14"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 5,
          "id": "Khmer",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/khmer-left.apng",
            "animated_right": "/images/units-animated/khmer-right.apng",
            "emblem": "/images/civemblems/khmer.png",
            "unit": "/images/civs/khmer.png",
          },
          "name": "Khmer",
        }
      }
      draftOptionPanelType="choice"
      key="15"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 2,
          "id": "Koreans",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/koreans-left.apng",
            "animated_right": "/images/units-animated/koreans-right.apng",
            "emblem": "/images/civemblems/koreans.png",
            "unit": "/images/civs/koreans.png",
          },
          "name": "Koreans",
        }
      }
      draftOptionPanelType="choice"
      key="16"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 3,
          "id": "Magyars",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/magyars-left.apng",
            "animated_right": "/images/units-animated/magyars-right.apng",
            "emblem": "/images/civemblems/magyars.png",
            "unit": "/images/civs/magyars.png",
          },
          "name": "Magyars",
        }
      }
      draftOptionPanelType="choice"
      key="17"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 5,
          "id": "Malay",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/malay-left.apng",
            "animated_right": "/images/units-animated/malay-right.apng",
            "emblem": "/images/civemblems/malay.png",
            "unit": "/images/civs/malay.png",
          },
          "name": "Malay",
        }
      }
      draftOptionPanelType="choice"
      key="18"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 4,
          "id": "Malians",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/malians-left.apng",
            "animated_right": "/images/units-animated/malians-right.apng",
            "emblem": "/images/civemblems/malians.png",
            "unit": "/images/civs/malians.png",
          },
          "name": "Malians",
        }
      }
      draftOptionPanelType="choice"
      key="19"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 2,
          "id": "Mayans",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/mayans-left.apng",
            "animated_right": "/images/units-animated/mayans-right.apng",
            "emblem": "/images/civemblems/mayans.png",
            "unit": "/images/civs/mayans.png",
          },
          "name": "Mayans",
        }
      }
      draftOptionPanelType="choice"
      key="20"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 1,
          "id": "Mongols",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/mongols-left.apng",
            "animated_right": "/images/units-animated/mongols-right.apng",
            "emblem": "/images/civemblems/mongols.png",
            "unit": "/images/civs/mongols.png",
          },
          "name": "Mongols",
        }
      }
      draftOptionPanelType="choice"
      key="21"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 1,
          "id": "Persians",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/persians-left.apng",
            "animated_right": "/images/units-animated/persians-right.apng",
            "emblem": "/images/civemblems/persians.png",
            "unit": "/images/civs/persians.png",
          },
          "name": "Persians",
        }
      }
      draftOptionPanelType="choice"
      key="22"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 4,
          "id": "Portuguese",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/portuguese-left.apng",
            "animated_right": "/images/units-animated/portuguese-right.apng",
            "emblem": "/images/civemblems/portuguese.png",
            "unit": "/images/civs/portuguese.png",
          },
          "name": "Portuguese",
        }
      }
      draftOptionPanelType="choice"
      key="23"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 1,
          "id": "Saracens",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/saracens-left.apng",
            "animated_right": "/images/units-animated/saracens-right.apng",
            "emblem": "/images/civemblems/saracens.png",
            "unit": "/images/civs/saracens.png",
          },
          "name": "Saracens",
        }
      }
      draftOptionPanelType="choice"
      key="24"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 3,
          "id": "Slavs",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/slavs-left.apng",
            "animated_right": "/images/units-animated/slavs-right.apng",
            "emblem": "/images/civemblems/slavs.png",
            "unit": "/images/civs/slavs.png",
          },
          "name": "Slavs",
        }
      }
      draftOptionPanelType="choice"
      key="25"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 2,
          "id": "Spanish",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/spanish-left.apng",
            "animated_right": "/images/units-animated/spanish-right.apng",
            "emblem": "/images/civemblems/spanish.png",
            "unit": "/images/civs/spanish.png",
          },
          "name": "Spanish",
        }
      }
      draftOptionPanelType="choice"
      key="26"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 1,
          "id": "Teutons",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/teutons-left.apng",
            "animated_right": "/images/units-animated/teutons-right.apng",
            "emblem": "/images/civemblems/teutons.png",
            "unit": "/images/civs/teutons.png",
          },
          "name": "Teutons",
        }
      }
      draftOptionPanelType="choice"
      key="27"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 1,
          "id": "Turks",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/turks-left.apng",
            "animated_right": "/images/units-animated/turks-right.apng",
            "emblem": "/images/civemblems/turks.png",
            "unit": "/images/civs/turks.png",
          },
          "name": "Turks",
        }
      }
      draftOptionPanelType="choice"
      key="28"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 5,
          "id": "Vietnamese",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/vietnamese-left.apng",
            "animated_right": "/images/units-animated/vietnamese-right.apng",
            "emblem": "/images/civemblems/vietnamese.png",
            "unit": "/images/civs/vietnamese.png",
          },
          "name": "Vietnamese",
        }
      }
      draftOptionPanelType="choice"
      key="29"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 1,
          "id": "Vikings",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/vikings-left.apng",
            "animated_right": "/images/units-animated/vikings-right.apng",
            "emblem": "/images/civemblems/vikings.png",
            "unit": "/images/civs/vikings.png",
          },
          "name": "Vikings",
        }
      }
      draftOptionPanelType="choice"
      key="30"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 6,
          "id": "Bulgarians",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/bulgarians-left.apng",
            "animated_right": "/images/units-animated/bulgarians-right.apng",
            "emblem": "/images/civemblems/bulgarians.png",
            "unit": "/images/civs/bulgarians.png",
          },
          "name": "Bulgarians",
        }
      }
      draftOptionPanelType="choice"
      key="31"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 6,
          "id": "Cumans",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/cumans-left.apng",
            "animated_right": "/images/units-animated/cumans-right.apng",
            "emblem": "/images/civemblems/cumans.png",
            "unit": "/images/civs/cumans.png",
          },
          "name": "Cumans",
        }
      }
      draftOptionPanelType="choice"
      key="32"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 6,
          "id": "Lithuanians",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/lithuanians-left.apng",
            "animated_right": "/images/units-animated/lithuanians-right.apng",
            "emblem": "/images/civemblems/lithuanians.png",
            "unit": "/images/civs/lithuanians.png",
          },
          "name": "Lithuanians",
        }
      }
      draftOptionPanelType="choice"
      key="33"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 6,
          "id": "Tatars",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/tatars-left.apng",
            "animated_right": "/images/units-animated/tatars-right.apng",
            "emblem": "/images/civemblems/tatars.png",
            "unit": "/images/civs/tatars.png",
          },
          "name": "Tatars",
        }
      }
      draftOptionPanelType="choice"
      key="34"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 7,
          "id": "Burgundians",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/burgundians-left.apng",
            "animated_right": "/images/units-animated/burgundians-right.apng",
            "emblem": "/images/civemblems/burgundians.png",
            "unit": "/images/civs/burgundians.png",
          },
          "name": "Burgundians",
        }
      }
      draftOptionPanelType="choice"
      key="35"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 7,
          "id": "Sicilians",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/sicilians-left.apng",
            "animated_right": "/images/units-animated/sicilians-right.apng",
            "emblem": "/images/civemblems/sicilians.png",
            "unit": "/images/civs/sicilians.png",
          },
          "name": "Sicilians",
        }
      }
      draftOptionPanelType="choice"
      key="36"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 8,
          "id": "Bohemians",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/bohemians-left.apng",
            "animated_right": "/images/units-animated/bohemians-right.apng",
            "emblem": "/images/civemblems/bohemians.png",
            "unit": "/images/civs/bohemians.png",
          },
          "name": "Bohemians",
        }
      }
      draftOptionPanelType="choice"
      key="37"
    />
    <Connect(Component)
      active={false}
      draftOption={
        Civilisation {
          "gameVersion": 8,
          "id": "Poles",
          "imageUrls": Object {
            "animated_left": "/images/units-animated/poles-left.apng",
            "animated_right": "/images/units-animated/poles-right.apng",
            "emblem": "/images/civemblems/poles.png",
            "unit": "/images/civs/poles.png",
          },
          "name": "Poles",
        }
      }
      draftOptionPanelType="choice"
      key="38"
    />
  </div>
</div>
`;
